src/search_renderer.cpp

```diff
@@ -93,30 +93,38 @@ std::string SearchRenderer::getHtml()
   kainjow::mustache::data pages{kainjow::mustache::data::type::list};
 
   auto resultStart = mp_searcher->getResultStart();
-  auto resultEnd = mp_searcher->getResultEnd();
-  auto resultCountPerPage = resultEnd - resultStart;
+  auto pageLength = this->pageLength;
   auto estimatedResultCount = mp_searcher->getEstimatedResultCount();
   auto currentPage = 0U;
   auto pageStart = 0U;
   auto pageEnd = 0U;
   auto lastPageStart = 0U;
-  if (resultCountPerPage) {
-    currentPage = resultStart/resultCountPerPage;
+  if (pageLength) {
+    currentPage = resultStart/pageLength;
     pageStart = currentPage > 4 ? currentPage-4 : 0;
     pageEnd = currentPage + 5;
-    if (pageEnd > estimatedResultCount / resultCountPerPage) {
-      pageEnd = estimatedResultCount / resultCountPerPage;
+    if (pageEnd > estimatedResultCount / pageLength) {
+      pageEnd = estimatedResultCount / pageLength;
+      //to include last page in pagination if page is not empty
+      if(pageEnd*pageLength!=estimatedResultCount)
+      {
+        pageEnd++;
+      }
```

Comment:
So you want to find the upward rounded result of `estimatedResultCount / pageLength` (i.e. the smallest value of `pageEnd` such that `pageEnd*pageLength>=estimatedResultCount`). This can be computed with a single expression as follows:
```
pageEnd = (estimatedResultCount + pageLength - 1) / pageLength;
```